src/Chapter1.hs

```diff
@@ -641,8 +650,10 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
-
+firstDigit n = 
+    if (n>=10)
+      then firstDigit (div n 10)
+      else n
```

Comment:
Your implementation is almost correct, but it returns wrong results on negative numbers. Probably you can patch it to take this corner case into consideration :slightly_smiling_face:

Reply:
I almost forgot considering negative numbers! @chshersh 
Thank you for reviewing my code! :D